src/VecchiaMLE_kernels.jl

```diff
@@ -1,18 +1,45 @@
-# We want to replace this function by a kernel implemented with KernelAbstractions.jl
-function vecchia_mul!(y::CuVector{T}, B::Vector{<:CuMatrix{T}}, x::CuVector{T}, n::Int, m::Vector{Int}) where T <: AbstractFloat
-    # replace with kernel from KernelAbstractions.jl. 
-    pos = 0
-    for j = 1:n
-        Bj = B[j]
-        yj = view(y, pos+1:pos+m[j])
-        xj = view(x, pos+1:pos+m[j])
-        mul!(yj, Bj, xj)
-        pos = pos + m[j]
+@kernel function vecchia_mul_kernel!(y, @Const(hess_obj_vals), @Const(x), @Const(m), @Const(n), @Const(colptrL), @Const(offsets))
+    index = @index(Global)
+
+    pos = colptrL[index]
+    offset = offsets[index]
+    mj = m[index]
```

Comment:
What may be faster is doing `mj = maximum(m)` outside the kernel and as `Val(mj)` and then `if j<mj`. Same for i. Just suggesting if this is too slow.

Reply:
The code is not working right now @michel2323 and I don't understand now. :(